scripts/camViewer

```diff
@@ -258,15 +274,44 @@ fi
 
 if [ $ENABLE -gt 0 ]; then 
     echo "Enabling ${CAMNAME}"
-    IOC=$(iocfpv "${CAMPVFULL}")
-    $imgr "$IOC" --hutch "$hutch" --enable   
+    for ioc in `iocfpv "$CAMPVFULL:IMAGE1"`; do
+        
+        INFO=$(grep_ioc "$ioc" all | grep "id:'$ioc'")
+        if [ -z "$INFO" ]; then
+             
+            >&2
+        else
+            
+            IOC=$(echo "$INFO" | sed -n "s/^.*id: '\(\S*\)'.*$/\1/p")
+            IOCHOST=$(echo "$INFO" | sed -n "s/^.*host: '\(\S*\)'.*$/\1/p")
+
+            iochutch=$(echo $INFO | grep "/reg/g/pcds/pyps/config/*" | cut -d'/' -f7)
+
+        fi
+    done
+
+    imgr "$ioc" --hutch "$iochutch" --enable   
     exit
 fi
 
 if [ $DISABLE -gt 0 ]; then
     echo "Disabling ${CAMNAME}"
-    IOC=$(iocfpv "${CAMPVFULL}")
-    $imgr "$IOC" --hutch "$hutch" --disable    
+    for ioc in `iocfpv "$CAMPVFULL:IMAGE1"`; do
```

Comment:
same comments/questions about the second loop as for the first loop, but also perhaps there was a way to combine these two into one section to avoid code duplication?

Reply:
I will attempt to create a function that can be called in both.